dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusPTX.java

```diff
@@ -0,0 +1,310 @@
+        programBuilder.initRegEqConst(ctx.threadId().id, ctx.register().getText(), new IValue(new BigInteger(ctx.constant().getText()), getArchPrecision()));
+        return null;
+    }
+
+    @Override
+    public Object visitVariableDeclaratorRegisterLocation(LitmusPTXParser.VariableDeclaratorRegisterLocationContext ctx) {
+        programBuilder.initRegEqLocPtr(ctx.threadId().id, ctx.register().getText(), ctx.location().getText(), getArchPrecision());
+        return null;
+    }
+
+    @Override
+    public Object visitVariableDeclaratorLocationLocation(LitmusPTXParser.VariableDeclaratorLocationLocationContext ctx) {
+        programBuilder.initLocEqLocPtr(ctx.location(0).getText(), ctx.location(1).getText());
+        return null;
+    }
+
+    // ----------------------------------------------------------------------------------------------------------------
+    // Proxy declarator list
+    @Override
+    public Object visitVariableDeclaratorProxy(LitmusPTXParser.VariableDeclaratorProxyContext ctx) {
+        if (ctx.proxyType().content.equals(Tag.PTX.GEN)) {
+            programBuilder.initLocEqLocAliasGen(ctx.location(0).getText(),
+                    ctx.location(1).getText());
+        } else {
+            programBuilder.initLocEqLocAliasProxy(ctx.location(0).getText(),
+                    ctx.location(1).getText());
+        }
+        return null;
+    }
+
+    // ----------------------------------------------------------------------------------------------------------------
+    // Thread declarator list (on top of instructions)
+    @Override
+    public Object visitThreadDeclaratorList(LitmusPTXParser.ThreadDeclaratorListContext ctx) {
+        for (LitmusPTXParser.ThreadScopeContext threadScopeContext : ctx.threadScope()) {
+            int ctaID = threadScopeContext.scopeID().ctaID().id;
+            int gpuID = threadScopeContext.scopeID().gpuID().id;
+            programBuilder.initScopedThread(threadScopeContext.threadId().id, ctaID, gpuID);
+            threadCount++;
+        }
+        return null;
+    }
+
+    // ----------------------------------------------------------------------------------------------------------------
+    // Instruction list (the program itself)
+    @Override
+    public Object visitConstant(LitmusPTXParser.ConstantContext ctx) {
+        return new IValue(new BigInteger(ctx.getText()), getArchPrecision());
+    }
+
+    @Override
+    public Object visitInstructionRow(LitmusPTXParser.InstructionRowContext ctx) {
+        for (int i = 0; i < threadCount; i++) {
+            mainThread = i;
+            visitInstruction(ctx.instruction(i));
+        }
+        return null;
+    }
+
+    @Override
+    public Object visitStoreConstant(LitmusPTXParser.StoreConstantContext ctx) {
+        MemoryObject object = programBuilder.getOrNewObject(ctx.location().getText());
+        IValue constant = new IValue(new BigInteger(ctx.constant().getText()), getArchPrecision());
```

Comment:
this can probably be replaced by `(IConst) ctx.constant().accept(this)`

Reply:
Replaced